draft-ietf-quic-recovery.md

```diff
@@ -699,6 +703,9 @@ Pseudocode for OnAckReceived and UpdateRtt follow:
     largest_acked_packet = max(largest_acked_packet,
                                ack.largest_acked)
 
+    if (from Handshake packet):
```

Comment:
```suggestion
    if (ack from Handshake packet):
```

Reply:
Can you add a short comment like "Once the client receives an ACK in a Handshake packet, it knows the server considers its address verified."?

Reply:
> The client can know that it is unverified if it is sending Initial packets and not Handshake (or greater).

That's not true, though: the client doesn't know it's been validated until it's seen an ACK using Handshake keys (or greater). The server needs to see a Handshake packet to validate the client, and until the client's seen such an ACK it can't be sure that any Handshake packets it might have sent so far weren't lost.

> As for the test in SetLossDetectionTimer() or LossDetectionTimeout(), how is it possible for a crypto packet to be in flight, where this condition would be true?

I'm not sure I understand the question. The tests in those functions are to ensure that a client sends packets to either earn anti-amplification credit or validate its address even when it has no information to send.

Reply:
Are you suggesting that we ditch `unverified_client` in favor of checking whether the endpoint is a client that doesn't yet have 1-RTT? I think that works as a simplification, if I'm understanding correctly.

Reply:
I think that works.  For the most part "do I have crypto packets in flight" covers the rest of it.  Though "in flight" might be wrong, and it might be better as "are any crypto packets that I have sent, but are unacknowledged".

The cost is that the client might keep this timer well past the point it is needed when acknowledgments are lost, but the simplification is pretty big.